### PR TITLE
Allow creation and deletion of ad hoc jobs along with exec and file copying

### PR DESCRIPTION
**Pull request by echee-wish**

### CODE REVIEW

**Comment by mlumibao-wish** [comment]
Changes: Removed `ctl run`
Added these new functions:
**ctl up APPNAME [flags]** - checks the ctl config file for the key APPNAME and _kubectl apply_ the manifest attached to it. This will spawn the ad hoc job along with its associated pods. If there are current jobs running, it will delete those jobs first before creating the new one.

**ctl down APPNAME** - finds all adhoc jobs created by the user and deletes them.

**ctl login APPNAME** - finds adhoc job created by the user and will shell into its pod. Similar to **ctl sh** but is intentionally made for ad hoc jobs

**ctl cp in/out POD SOURCE [flags]** - will copy files from local to pod (or vice versa) to the pod (first container used if no container is specified). Default destination is /tmp/ctl unless specified by _out_ flag. Uses _kubectl cp_.

**Comment by echee-wish** [comment]
run `golint ./...` in the project directory and resolve the warinings

File: README.md [hunk 1]
```diff
@@ -117,9 +116,9 @@ A big feature of ctl is the support for multiple logs at a time. As all of ctl u
 Additionally, this feature supports the `-f, --follow` option which constantly streams logs.
 
 ## Run
-The run command is a wrapper over `kubectl run`. It requires cluster config to set up.
+The run command is a wrapper over `kubectl apply`. It requires cluster config to set up.
 
-The syntax is `ctl run NAME [flags]`. The label flags can be used to narrow down which cluster to run on. If there are multiple clusters in which a run is specified on, ctl randomly picks one.
+The syntax is `ctl run APPNAME [flags]`. The label flags can be used to narrow down which cluster to run on. If there are multiple clusters in which a run is specified on, ctl randomly picks one.
```
Comment (echee-wish) [inline README.md]: didn't we change `run` to `up`

**Comment by echee-wish** [comment]
Because we use the machine's hostname to name the jobs, if someone does `ctl` up on their macbook and later tries to do `ctl cp` on their linux machine to get their script data it wont work. we should should either add a flag like `--name or -n` to override the hostname or possible find a way to get their username

File: README.md (hunk 1 above)
Comment (mlumibao-wish) [reply README.md]: Still need to update the readme, was going to do it after everything is approved

**Comment by mlumibao-wish** [comment]
> Because we use the machine's hostname to name the jobs, if someone does ctl up on their macbook and later tries to do ctl cp on their linux machine to get their script data it wont work. we should should either add a flag like --name or -n to override the hostname or possible find a way to get their username

Yeah adding a flag --user -u should work to override that

**Pull request merged**